database/rrd.c

```diff
@@ -166,3 +80,355 @@ STRING *rrd_string_strdupz(const char *s) {
     freez(tmp);
     return ret;
 }
+
+// ----------------------------------------------------------------------------
+// rrd global / startup initialization
+
+#ifdef ENABLE_DBENGINE
+struct dbengine_initialization {
+    netdata_thread_t thread;
+    char path[FILENAME_MAX + 1];
+    int disk_space_mb;
+    size_t tier;
+    int ret;
+};
+
+static void *dbengine_tier_init(void *ptr) {
+    struct dbengine_initialization *dbi = ptr;
+    dbi->ret = rrdeng_init(NULL, dbi->path, dbi->disk_space_mb, dbi->tier);
+    return ptr;
+}
+#endif
+
+static void dbengine_init(const char *hostname) {
+#ifdef ENABLE_DBENGINE
+    rrdb.use_direct_io = config_get_boolean(CONFIG_SECTION_DB, "dbengine use direct io", rrdb.use_direct_io);
+
+    unsigned read_num = (unsigned)config_get_number(CONFIG_SECTION_DB, "dbengine pages per extent", MAX_PAGES_PER_EXTENT);
+    if (read_num > 0 && read_num <= MAX_PAGES_PER_EXTENT)
+        rrdeng_pages_per_extent = read_num;
+    else {
+        netdata_log_error("Invalid dbengine pages per extent %u given. Using %u.", read_num, rrdeng_pages_per_extent);
+        config_set_number(CONFIG_SECTION_DB, "dbengine pages per extent", rrdeng_pages_per_extent);
+    }
+
+    rrdb.storage_tiers = config_get_number(CONFIG_SECTION_DB, "storage tiers", rrdb.storage_tiers);
+    if (rrdb.storage_tiers < 1) {
+        netdata_log_error("At least 1 storage tier is required. Assuming 1.");
+        rrdb.storage_tiers = 1;
+        config_set_number(CONFIG_SECTION_DB, "storage tiers", rrdb.storage_tiers);
+    }
+    if (rrdb.storage_tiers > RRD_STORAGE_TIERS) {
+        netdata_log_error("Up to %d storage tier are supported. Assuming %d.", RRD_STORAGE_TIERS, RRD_STORAGE_TIERS);
+        rrdb.storage_tiers = RRD_STORAGE_TIERS;
+        config_set_number(CONFIG_SECTION_DB, "storage tiers", rrdb.storage_tiers);
+    }
+
+    bool parallel_initialization = (rrdb.storage_tiers <= (size_t)get_netdata_cpus()) ? true : false;
+    parallel_initialization = config_get_boolean(CONFIG_SECTION_DB, "dbengine parallel initialization", parallel_initialization);
+
+    struct dbengine_initialization tiers_init[RRD_STORAGE_TIERS] = {};
+
+    size_t created_tiers = 0;
+    char dbenginepath[FILENAME_MAX + 1];
+    char dbengineconfig[200 + 1];
+    int divisor = 1;
+    for(size_t tier = 0; tier < rrdb.storage_tiers ;tier++) {
+        if(tier == 0)
+            snprintfz(dbenginepath, FILENAME_MAX, "%s/dbengine", netdata_configured_cache_dir);
+        else
+            snprintfz(dbenginepath, FILENAME_MAX, "%s/dbengine-tier%zu", netdata_configured_cache_dir, tier);
+
+        int ret = mkdir(dbenginepath, 0775);
```

Comment:
This is outside the scope of this PR. The code has been copied/pasted from `rrdhost.c`. This would be a functional change and a PITA to debug if something goes wrong.